controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -2005,7 +2039,7 @@ func TestInvalidBMHCanBeDeleted(t *testing.T) {
 	doDeleteHost(host, r)
 
 	tryReconcile(t, r, host, func(host *metal3v1alpha1.BareMetalHost, result reconcile.Result) bool {
-		return host.Status.Provisioning.State == metal3v1alpha1.StateDeleting && len(host.Finalizers) == 0
+		return host.UID == "" && len(host.Finalizers) == 0
```

Comment:
That's interesting. There was a breaking change in the newer version of  `fakeClient`, since now it removes effectively from its internal cache a deleted object, introduced in the [controller-runtime PR #139](https://github.com/kubernetes-sigs/controller-runtime/pull/1399). 
By reviewing your comment I realized that very likely this fact couldn't be very clear for a test writer/reader, so I updated the `tryReconcile` code to nullify the host when not found in the cache, so that hopefully the contract will be clearer for the `isDone` part